python/cudf/cudf/core/series.py

```diff
@@ -857,45 +859,47 @@ def map(self, arg, na_action=None) -> "Series":
         3      NaN
         dtype: object
 
-        It also accepts a function:
-
-        >>> s.map('I am a {}'.format)
-        0       I am a cat
-        1       I am a dog
-        2       I am a nan
-        3    I am a rabbit
+        It also accepts numeric functions:
+        >>> s = cudf.Series([1, 2, 3, 4, np.nan])
+        >>> s.map(lambda x: x ** 2)
+        0       1
+        1       4
+        2       9
+        3       16
         dtype: object
 
-        To avoid applying the function to missing values (and keep them as
-        ``NaN``) ``na_action='ignore'`` can be used:
-
-        >>> s.map('I am a {}'.format, na_action='ignore')
-        0     I am a cat
-        1     I am a dog
-        2            NaN
-        3  I am a rabbit
-        dtype: object
+        Please note map does not currently support user defined functions
+        with dtypes `str` and`category`.
         """
         if isinstance(arg, dict):
-            self_df = self.to_frame()
-            dataframe_1 = cudf.merge(
-                cudf.Series(arg).to_frame(), self_df, how="left", sort=True
+            lhs = cudf.DataFrame({"x": self, "orig_order": arange(len(self))})
+            rhs = cudf.DataFrame(
+                {
+                    "x": arg.keys(),
+                    "s": arg.values(),
+                    "bool": full(len(arg), True, dtype=self.dtype),
+                }
             )
-            dataframe_2 = self_df.join(
-                dataframe_1, lsuffix="_caller", rsuffix="_other", how="left"
+            res = lhs.merge(rhs, on="x", how="left").sort_values(
+                by="orig_order"
             )
-            result = dataframe_2["0_other"]
+            res = res.drop_duplicates(subset="orig_order", ignore_index=True)
```

Comment:
How could we end up with duplicates here? Since `arg` is a dict it guarantees `arg.keys()` is unique, which means there can only be at most one match per row in the `lhs` dataframe.

Reply:
I think in this case the drop was to remove the index from orig_order. Once I change the index as you suggested it gives the correct result :D I'll push the updated changes now.